service/vector_store_client.cc

```diff
@@ -552,10 +560,81 @@ struct vector_store_client::impl {
 
         co_return std::unexpected{service_unavailable{}};
     }
+
+    auto ann(keyspace_name keyspace, index_name name, schema_ptr schema, vs_vector vs_vector, limit limit, abort_source& as)
+            -> future<std::expected<primary_keys, ann_error>> {
+        if (is_disabled()) {
+            vslogger.error("Disabled Vector Store while calling ann");
+            co_return std::unexpected{disabled{}};
+        }
+
+        auto path = format("/api/v1/indexes/{}/{}/ann", keyspace, name);
+        auto content = write_ann_json(std::move(vs_vector), limit);
+
+        auto resp = co_await make_request(operation_type::POST, std::move(path), std::move(content), as);
+        if (!resp) {
+            co_return std::unexpected{std::visit(
+                    [](auto&& err) {
+                        return ann_error{err};
+                    },
+                    resp.error())};
+        }
+
+        if (resp->status != status_type::ok) {
+            vslogger.error("Vector Store returned error: HTTP status {}: {}", resp->status,
+                        seastar::value_of([&resp] {return response_content_to_sstring(resp->content);}));
+            co_return std::unexpected{service_error{resp->status}};
+        }
+
+        try {
+            co_return read_ann_json(rjson::parse(std::move(resp->content)), schema);
+        } catch (const rjson::error& e) {
+            vslogger.error("Vector Store returned invalid JSON: {}", e.what());
+            co_return std::unexpected{service_reply_format_error{}};
+        }
+    }
+};
+
+class vector_store_client::metrics{
+    lw_shared_ptr<impl> _impl;
+    uint64_t dns_refreshes = 0;
+    utils::estimated_histogram ann_limit{35}; // 35 to cover the range from 1 to 1000
+    seastar::metrics::metric_groups _metrics;
+
+public:
+
+    metrics(lw_shared_ptr<impl> _impl)
+        : _impl(_impl) {
+        _metrics.add_group("vector_store", {
+                                                   seastar::metrics::make_gauge(
+                                                           "dns_refreshes",
+                                                           [this] {
+                                                               return dns_refreshes;
+                                                           },
+                                                           seastar::metrics::description("Number of DNS refreshes")),
+                                                   seastar::metrics::make_histogram("ann_limit", seastar::metrics::description("ANN request limit"),
+                                                           [this]() {
+                                                               return estimated_histogram_to_metrics(ann_limit);
```

Comment:
I thought we moved that to the vector store.
There is no timing involved; at the vector store level, we have it once per index, and we'll make this PR even simpler.